cg/services/order_validation_service/rules/case_sample/utils.py

```diff
@@ -0,0 +1,49 @@
+from cg.constants.constants import PrepCategory
+from cg.models.orders.sample_base import ContainerEnum
+from cg.services.order_validation_service.constants import MAXIMUM_VOLUME, MINIMUM_VOLUME
+from cg.services.order_validation_service.models.sample import Sample
+from cg.services.order_validation_service.workflows.tomte.models.sample import TomteSample
+from cg.store.models import Application
+from cg.store.store import Store
+
+
+def is_application_not_compatible(
+    allowed_prep_categories: list[PrepCategory],
+    application_tag: str,
+    store: Store,
+) -> bool:
+    application: Application = store.get_application_by_tag(application_tag)
+    return application and (application.prep_category not in allowed_prep_categories)
+
+
+def is_concentration_missing(sample: TomteSample) -> bool:
+    return not sample.concentration_ng_ul
+
+
+def is_well_position_missing(sample: TomteSample) -> bool:
+    return sample.container == ContainerEnum.plate and not sample.well_position
+
+
+def is_container_name_missing(sample: TomteSample) -> bool:
```

Comment:
I am a bit uncertain if this is a case_sample util or a sample util. Same goes for a lot of these I guess. Regardless of workflow I think you can specify container, definitely for some standalone sample workflows

Reply:
Currently, they are producing `CaseSampleErrors` and as such, they belong in the `case_sample` module. They also operate on Tomte specific types, which means that they belong in the tomte module. I'll move them there.

Reply:
That is not what I am after. This bool will need to be calculated for all workflows. As such the utils function (not the whole rule) could very well be placed in a different dir.

Reply:
I'll leave this for separate PR:s, as discussed in slack